stdlib/synch.ysh

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env ysh
+
+module stdlib/synch || return 0
+
+############################
+### FIFO File Desriptors ###
+############################
+
+func fifoFdNew() {
+  # WARN: this section should be critical but for now it's not
+  # A solution may be retry on fail.
+  #====================
+  var fifo = $(mktemp -u)
+  mkfifo $fifo
+  #====================
+  exec {fd}<>$fifo
+  return (fifo)
+}
+
+func fifoFdDestroy(fd) {
+  var fifoFile = $(readlink /proc/$$/fd/$fd)
+  exec {fd}>-
+  exec {fd}<-
+  rm $fifoFile
+}
+
+#################
+### Semaphore ###
+#################
+
+func semaNew(value) {
```

Comment:
Likewise, these should be procs, with kebab-case e.g.

https://www.oilshell.org/release/0.21.0/doc/style-guide.html

```
sema-new
sema-up
sema-down
```